platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/XMLUtils.java

```diff
@@ -124,28 +135,66 @@ public static Result transform(Node nodeXml, TransformerProperties transformerPr
      */
     public static String getRootNamespace(String xml) {
 
-        String rootNamespace = null;
-
         if (xml == null) {
             return null;
         }
 
-        initializeXMLInputFactory();
+        ProcessingResult<String> result = new ProcessingResult();
+        return processElements(xml, result, (xmlStreamReader) -> {
+            result.setValue(xmlStreamReader.getNamespaceURI());
+            result.done();
+        });
+    }
+
+    /**
+     * This method lets clients parse through the elements of an XML document with less
+     * boiler-plate code. This methods handles the configuration of the
+     * XML factory and disables security risks like macro expansion.
+     * This methods also catches the several exceptions that can be thrown during the creation
+     * and use of an XML parser.
+     * <p>
+     * Example: Get the first three elements whose local name is "Fizz".
+     * <p>
+     * String xml = "my XML document"; // The string representation of the XML document to parse
+     * <p>
+     * XMLUtils.ProcessingResult<List<String>> result =
+     * new XMLUtils.ProcessingResult<>(new ArrayList<String>());
+     * <p>
+     * List<String> output = XMLUtils.processElements(xml, result, (xmlReader) -> {
+     * if (xmlReader.getLocalName().equals("Fizz")) {
+     * if (result.getValue().size() < 3) {
+     * result.getValue().add(xmlReader.getElementText());
+     * } else { result.done(); }
+     * }
+     * });
+     *
+     * @param xml                    The XML to process
+     * @param result                 Instance of ProcessingResult
+     * @param processElementFunction Function that accepts an instance of XMLStreamReader
+     *                               and updates the processing result
+     * @param <T>                    The type of the processing result's value
+     * @return Processing results's value
+     */
+    public static <T> T processElements(String xml, ProcessingResult<T> result,
+            Consumer<XMLStreamReader> processElementFunction) {
 
+        initializeXMLInputFactory();
         XMLStreamReader xmlStreamReader = null;
 
         try (StringReader strReader = new StringReader(xml)) {
             xmlStreamReader = xmlInputFactory.createXMLStreamReader(strReader);
-
             while (xmlStreamReader.hasNext()) {
                 int event = xmlStreamReader.next();
                 if (event == XMLStreamConstants.START_ELEMENT) {
-                    rootNamespace = xmlStreamReader.getNamespaceURI();
-                    return rootNamespace;
+                    processElementFunction.accept(xmlStreamReader);
+                    if (result.isDone) {
+                        break;
+                    }
                 }
             }
         } catch (XMLStreamException e) {
-            LOGGER.debug("Unable to parse root namespace from XML", e);
+            result.setValue(null);
```

Comment:
An exception will set the `ProcessingResults` value to `null`. Clients can check the value to see if processing terminated abnormally.